### ISSUE TITLE
About the permutation-invariant average precision (PIAP) metric

### ISSUE THREAD

**Issue by junwenxiong** [hxixixh/mix-and-localize#3]
Hi, Xixi. 
The calculation process of the PIAP is not described in detail in the paper.  How to compute PIAP in formula? Could you please provide some  mathematical calculations? 
Thanks.

**Comment by hxixixh** [comment]
Hi @woshihaozhu, 

Thanks for taking an interest in our work. 
Regarding the calculation for PIAP(AP), we used scikit-learn's implementation of average precision score.  You can see the code for the calculation below. 

```bash
def calc_ap(pred_prob, gt_masks): 
    y_true = gt_masks.reshape(-1)
    y_score = pred_prob.reshape(-1)
    ap = sklearn.metrics.average_precision_score(y_true, y_score)

    return ap
```
I hope this answers your questions.

**Comment by junwenxiong** [comment]
Thank you.  This answer cleared my doubts.
